docs/architecture/adr-000-generalize-genesis-accounts.md

```diff
@@ -0,0 +1,43 @@
+# ADR 000: Generalize Genesis Accounts
```

Comment:
This should exist as a separate PR first where we can discuss and agree on the high level approach before work continues on this PR 👍

Reply:
ok, I see format. New PR created here #4986 
I'll close this one.